### ISSUE TITLE
An CNN model error: You must feed a value for placeholder tensor 'y' with dtype int32 and shape [?] 

### ISSUE THREAD

**Issue by sunoranges** [tensorflow/tensorflow#28425]
import tensorflow as tf
import os
from skimage import io, transform
import glob

import numpy as np

#dataset parameter
mode = 'folder'
train_dataset_path = '/Users/resun/ML_data/train_data_png/'   #change here
test_dataset_path = '/Users/resun/ML_data/test_data_png/'
tag_dataset_path = '/Users/resun/ML_data/tag_image/'

'''
ROOT_FOLDER
       |-------- SUBFOLDER (CLASS 0)
       |             |
       |             | ----- image1.jpg
       |             | ----- image2.jpg
       |             | ----- etc...
       |             
       |-------- SUBFOLDER (CLASS 1)
       |             |
       |             | ----- image1.jpg
       |             | ----- image2.jpg
       |             | ----- etc...
'''
# image parameters
num_classes = 2     #change here
image_height = 128    #change here
image_width = 128    #change here
channels = 3     #the 3 color channels, change to 1 if grayscale
'''
0: Use the number of channels in the PNG-encoded image.
1: output a grayscale image.
3: output an RGB image.
4: output an RGBA image.
'''

#########################################
#########################################
#mode = folder
'''

def read_images(dataset_path, mode, batch_size):
    imagepaths, labels = list(), list()
    if mode == 'folder':
        # an ID will be affected to each sub-folders by alphabetical order
        label = 0
        #list the directory
        try:   #python2     #two paths in the dataset_path
            classes = sorted(os.walk(dataset_path).next()[1])
            #walk: For each directory in the directory tree rooted at top, yields a 3-tuple (dirpath, dirnames, filenames)
        except Exception:  #python3
            classes = sorted(os.walk(dataset_path).__next__()[1])

        #list each sub-directory (the classes)
        for c in classes:    #img inside the folder
            c_dir = os.path.join(dataset_path, c)
            try:   #python2
                walk = os.walk(c_dir).next()
            except Exception:   #python3
                walk = os.walk(c_dir).__next__()

            #add each image to the training set
            for sample in walk[2]:    #walk[2] in sub-directory
                #only keeps jpeg images
                if sample.endswith('.png') or sample.endswith('.jpeg'):
                    imagepaths.append(os.path.join(c_dir, sample))
                    labels.append(label)
            label += 1
    else:
        raise Exception("unknown mode")

    #img data convert to a tensor
    imagepaths = tf.convert_to_tensor(imagepaths, dtype = tf.string)
    labels = tf.convert_to_tensor(labels, dtype=tf.int32)
    #build a tf queue, shuffle data
    image, label = tf.train.slice_input_producer([imagepaths, labels], shuffle=True)   #Produces a slice of each `Tensor` in `tensor_list
    #read images from disk
    image = tf.read_file(image)    #Reads and outputs the entire contents of the input filename
    image = tf.image.decode_png(image, channels=channels)    #Decode a PNG-encoded image to a uint8 or uint16 tensor
    #resize images from disk
    image = tf.image.resize_images(image, [image_height, image_width])
    #normalize
    image = image * 1.0/127.5 - 1.0

    #create batches
    X, Y = tf.train.batch([image, label], batch_size = batch_size, capacity = batch_size*8, num_threads = 4)
    return X, Y
'''

def read_img(path):
    cate = [train_dataset_path+x for x in os.listdir(train_dataset_path) if os.path.isdir(train_dataset_path+x)]
    imgs = []
    labels = []
    for idx, folder in enumerate(cate):
        for im in glob.glob(folder+'\*.png'):
            #print('reading the images: %s' %(im))
            img = io.imread(im)
            img = transform.resize(img, (image_width, image_height, channels))
            imgs.append(img)
            labels.append(idx)
    X=np.asarray(imgs, np.float32)
    Y=np.asarray(labels, np.int32)
    return X, Y

#######################################
#######################################

#parameters
learning_rate = 0.001
num_steps = 100
batch_size = 50
display_steps = 50
dropout = 0.75    #probability to keep units

X_train, Y_train = read_img(train_dataset_path)
#X_test, Y_test = read_images(test_dataset_path, mode, batch_size)
#X_tag, Y_tag = read_images(tag_dataset_path, mode, batch_size)
print(X_train)
print(Y_train)
print(type(X_train))
print(type(Y_train))

#create model
x = tf.placeholder(tf.float32, shape=[None, image_width, image_height, channels], name = 'x')
y = tf.placeholder(tf.int32, shape=[None, ], name = 'y')
def conv_net(input, n_classes, dropout, reuse, training):
    #define a scope for reusing the variables
    with tf.variable_scope('ConvNet', reuse=reuse):
        conv1 = tf.layers.conv2d(input, 32, 5, activation=tf.nn.relu)   #convolution layer with 32 filters and a kernel size of 5
        conv1 = tf.layers.max_pooling2d(conv1, 2, 2)     #max pooling (down-sampling) with pool_size of 2 and strides(步长) of 2
        conv2 = tf.layers.conv2d(conv1, 64, 3, activation=tf.nn.relu)    #convolution layer with 64 filters and a kernel size of 3
        conv2 = tf.layers.max_pooling2d(conv2, 2, 2)     #max pooling (down-sampling) with pool-size of 2 and strides of 2
        fc1 = tf.contrib.layers.flatten(conv2)    #flatten the data to a 1D vector for the fully connected layer
        fc1 = tf.layers.dense(fc1, 1024)     #fully connected layer (in contrib folder for now)
        fc1 = tf.layers.dropout(fc1, rate = dropout, training = training)    #apply dropout (if training is false, dropout is not applied)
        out = tf.layers.dense(fc1, n_classes)     #output layer, class prediction
        out = tf.nn.softmax(out) if not training else out
    return out

constant = tf.constant(value=1, dtype=tf.float32)

logits_train = conv_net(x, num_classes, dropout, reuse=False, training=True)    #create a graph for training
logits_eval = tf.multiply(logits_train, constant, name='logits_eval')
#logits_test = conv_net(X_test, num_classes, dropout, reuse=True, training=False)    #create another graph for testing that reuse the same weights
#logits_tag = conv_net(X_tag, 2, dropout, reuse=True, training=False)

loss_op = tf.reduce_mean(tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits_train, labels=y))
optimizer = tf.train.AdamOptimizer(learning_rate=learning_rate)
train_op = optimizer.minimize(loss_op)
#evulate model
correct_pred = tf.equal(tf.argmax(tf.nn.softmax(logits_train), 1), tf.cast(y, tf.int64))
accuracy = tf.reduce_mean(tf.cast(correct_pred, tf.float32))
init = tf.global_variables_initializer()
#save object
saver = tf.train.Saver()   #The constructor adds ops to save and restore variables.

#########################################
#########################################

with tf.Session() as sess:

    sess.run(init)

    #start data queue
    coord = tf.train.Coordinator()
    tf.train.start_queue_runners(coord=coord)

    for step in range(1, num_steps+1):
        if step % display_steps == 0 or step == 1:    #run optimization and calculate batch loss and accuracy
            _, loss, acc = sess.run([train_op, loss_op, accuracy], feed_dict={x: X_train, y: Y_train})
            print("Step " + str(step) + ", Minibatch Loss= " + "{:.4f}".format(loss) + ", Training Accuracy= " + "{:.3f}".format(acc))
        else:
            sess.run(train_op)   #only run the optimization op (backprop)
    print("Optimization Finished!")

**Comment by sunoranges** [comment]
I think I have already give y placeholder a input Y_train, but still got error.

**Comment by achandraa** [comment]
Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?

Make sure you also include the exact command if possible to produce the output included in your test case. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by achandraa** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
